File: deaf-grandma.js
// Import implementation of prompt from library
let prompt = require('prompt');

let gbyes = 0;

function deafGrandma() {
  prompt.start(); // start prompt
  prompt.get('Say_something_to_Grandma', function(err, msg) { // Get input from Grandma's visitor
    if (gbyes < 2) {
      let note = msg.Say_something_to_Grandma; // Save message to a variable
      if (note == "" || note == null || note == undefined) {
        rant = 'WHAT!';
      } else if (/[a-z]/.test(note)) {
        rant = 'SPEAK UP, KID!'; 
      } else if (note == 'GOODBYE!') {
        if (gbyes == 0) {
          rant = 'LEAVING SO SOON?';
          gbyes++;
        } else if (gbyes == 1) {
          rant = 'LATER, SKATER!';
          gbyes++;
        }
      } else if (note === note.toUpperCase()) {
        rant = 'NO, NOT SINCE 1946!';
        }
      }
    console.log(rant);
    if (gbyes < 2) {
      deafGrandma();
    }
  })
}

deafGrandma();

File: deaf-grandma.py



def deaf_grandma():
  gbyes = 0
  msghaslower = False
  while (gbyes < 2):
    msg = input("Say something to Grandma: ")
    for char in msg:
      if char.islower():
        msghaslower = True
    if (msg == "" or msg == " "):
      rant = 'WHAT!'
    elif (msghaslower):
      rant = 'SPEAK UP, KID!'
      msghaslower = False
    elif (msg == 'GOODBYE!'):
      if (gbyes == 0):
        rant = 'LEAVING SO SOON?'
        gbyes = gbyes + 1
      elif (gbyes == 1):
        rant = 'LATER, SKATER!'
        gbyes = gbyes + 1
    elif (msg.isupper()):
      rant = 'NO, NOT SINCE 1946!'
    print(rant)

deaf_grandma()
